tests/user_examples/extra_src/shared/general_continuum/general_continuum.cpp

```diff
@@ -52,14 +52,15 @@ namespace SPH
         Real stress_tensor_I1 = stress_tensor.trace(); // first invariant of stress
         Mat3d deviatoric_stress_tensor = stress_tensor - (1.0 / dim) * stress_tensor.trace() * Mat3d::Identity();
         Real stress_tensor_J2 = 0.5 * (deviatoric_stress_tensor.cwiseProduct(deviatoric_stress_tensor.transpose())).sum();
-        Real f = sqrt(stress_tensor_J2) + alpha_fai_ * stress_tensor_I1 - k_c_;
+        Real f = sqrt(stress_tensor_J2) + alpha_phi_ * stress_tensor_I1 - k_c_;
         Real lambda_dot_ = 0;
         Mat3d g = Mat3d::Zero();
         if ((f >= TinyReal) && (stress_tensor_J2 > TinyReal))
         {
             Real deviatoric_stress_times_strain_rate = (deviatoric_stress_tensor.cwiseProduct(strain_rate)).sum();
-            lambda_dot_ = (3 * alpha_fai_ * K_ * strain_rate.trace() + (G_ / sqrt(stress_tensor_J2)) * deviatoric_stress_times_strain_rate) / (27 * alpha_fai_ * K_ * sin(psi_) + G_);
-            g = lambda_dot_ * (9 * K_ * sin(psi_) * Mat3d::Identity() + G_ * deviatoric_stress_tensor / (sqrt(stress_tensor_J2)));
+            //non-associate flow rule
+            lambda_dot_ = (3 * alpha_phi_ * K_ * strain_rate.trace() + (G_ / sqrt(stress_tensor_J2)) * deviatoric_stress_times_strain_rate) / (9 * alpha_phi_ * K_ * getDPConstantsA(psi_) + G_);
+            g = lambda_dot_ * (3 * K_ * getDPConstantsA(psi_) * Mat3d::Identity() + G_ * deviatoric_stress_tensor / (sqrt(stress_tensor_J2)));
```

Comment:
getDPConstantsA? why A?

Reply:
There are two constants in the DP constitutive model, and A is one of them.